Mini-Project 3: Text Mining

YehEmily opened the pull request:
Here's my work for MP3: a doge-style poem generator!

(For your reference: http://knowyourmeme.com/memes/doge)

(Also, I'm so sorry this is late; I missed all of Mini-Project 3 because I was in the hospital, and I decided to finish it after Mini-Project 4, so that I wouldn't get any more behind... Thank you for your patience and understanding, and I'm sorry for any inconveniences this might have caused!)

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/sd16spring/textmining/66)

<!-- Reviewable:end -->

YehEmily committed:
Submitting my text-mining project: a doge-style poem generator!

(Sorry this is late; I missed all of Mini-Project 3 because I was in the hospital, and I decided to finish it after Mini-Project 4, so that I wouldn't get any more behind... Thank you for your patience and understanding, and again, I'm sorry this is super late!)

YehEmily committed:
Here's the Shakespeare text that I used for this project!

YehEmily committed:
Here's my write-up and reflection!

jenwei commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_[doge_poetry.py, line 29 [r1]](https://reviewable.io:443/reviews/sd16spring/textmining/66#-KDOxXeK1DvBaRWqH8WP:-KDOxXeL5TfMH8edT5yY:1686874174) ([raw file](https://github.com/sd16spring/textmining/blob/a638858267bc6e3c2580847571fd7149e27ebabe/doge_poetry.py#L29)):_
This looks fine, but this all seems quite specific to project gutenberg, so if you ever revisit this, you could potentially take all the code here and create a `clean_gutenberg_text` function, and once the text is cleaned up, take that input and use `get_word_list`.

---

_[doge_poetry.py, line 66 [r1]](https://reviewable.io:443/reviews/sd16spring/textmining/66#-KDOxywguWGFxJWuaczC:-KDOxywguWGFxJWuaczD:-993027568) ([raw file](https://github.com/sd16spring/textmining/blob/a638858267bc6e3c2580847571fd7149e27ebabe/doge_poetry.py#L66)):_
Nice use of a negative index here to search from the back!

---

_[doge_poetry.py, line 84 [r1]](https://reviewable.io:443/reviews/sd16spring/textmining/66#-KDOy7PPXT_qOrSES9tD:-KDOy7PPXT_qOrSES9tE:116172376) ([raw file](https://github.com/sd16spring/textmining/blob/a638858267bc6e3c2580847571fd7149e27ebabe/doge_poetry.py#L84)):_
Great notes here about why you chose the `randint` ranges you did.

---

_[MP3_Reflection.pdf, line 0 [r1]](https://reviewable.io:443/reviews/sd16spring/textmining/66#-KDOwuhQDly46_Kd3gFl:-KDOwuhRbiARzc3SaezS:796457219) ([raw file](https://github.com/sd16spring/textmining/blob/a638858267bc6e3c2580847571fd7149e27ebabe/MP3_Reflection.pdf#L0)):_
Awesome writeup. Seems like you put a lot of thought into it, had a lot of fun, and learned quite a bit too. Those poems were quite entertaining. 

Nice that you tied in what you learned from the Word Frequency toolbox and applied it here.

The stanza issue is definitely tricky, but in your situation, hardcoding a skeleton seems like a fine idea.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sd16spring/textmining/66)_

<!-- Sent from Reviewable.io -->

YehEmily committed:
Updated for mini-project 5!

YehEmily committed:
Updated again for mini-project 5!

YehEmily committed:
Updated one last time for mini-project 5!

I kept forgetting to include comments that might help someone understand my code... here's the final version!
